src/ert/shared/storage/extraction.py

```diff
@@ -60,56 +61,47 @@ def create_parameters(ert) -> List[dict]:
 
 def _create_response_observation_links(ert) -> Mapping[str, str]:
     observation_vectors = ert.get_observations()
-    keys = [ert.get_observation_key(i) for i, _ in enumerate(observation_vectors)]
     summary_obs_keys = observation_vectors.getTypedKeylist(
         EnkfObservationImplementationType.SUMMARY_OBS
     )
-    if keys == []:
-        return {}
-
-    data = MeasuredData(ert, keys, load_data=False)
-    observations = data.data.loc[["OBS", "STD"]]
     response_observation_link = {}
 
-    for obs_key in observations.columns.get_level_values(0).unique():
-        obs_vec = observation_vectors[obs_key]
-        data_key = obs_vec.getDataKey()
-
+    for obs_vector in observation_vectors:
+        obs_key = obs_vector.getKey()
+        data_key = obs_vector.getDataKey()
+        # All summary observations for a specific summary vector are grouped
         if obs_key not in summary_obs_keys:
             response_observation_link[data_key] = obs_key
         else:
             response_observation_link[data_key] = data_key
+
     return response_observation_link
 
 
 def create_response_records(ert, ensemble_name: str, observations: List[dict]):
```

Comment:
As we have the concept of a `Record` in ERT, should this function return a list of such `Record`'s, or does `record` in this context refer to something different?

Reply:
This should probably be refactored, but not sure if this PR is the right time